### ISSUE TITLE
new option for Rooms: "allow only users with password and email set"

### ISSUE THREAD

**Issue by rubo77** [vector-im/riot-web#5170]
In many invite-only rooms it is essential, that users keep their account running, so an extra option besides "invite-only" would be needed to allow only users to access the room, if they have setup a password and email (maybe in two separate options)

**Comment by t3chguy** [comment]
This is unenforceable, anyone could modify synapse to ignore this rule and other servers would have no way of enforcing it as they cannot see a user's email//password if they are on a different server...

**Comment by rubo77** [comment]
ok, maybe then "allow only users from this server and with a password and email set"

**Comment by uhoreg** [comment]
Maybe doable with groups, if the server creates a group and automatically populates it with registered users.

**Comment by lampholder** [comment]
Yeah - it'd be a little way away (pending proper groups + acls based on said groups, etc) but I think I would favour a groups-based solution.
